primitives/contract/src/mapping_token_factory/s2s.rs

```diff
@@ -34,13 +34,18 @@ pub struct Sub2SubMappingTokenFactory;
 impl Sub2SubMappingTokenFactory {
 	/// encode confirm burn and remote unlock deliver message function
 	pub fn encode_confirm_burn_and_remote_unlock(
-		message_id: Vec<u8>,
+		lane_id: &[u8; 4],
```

Comment:
Recommend to use primitive type `LaneId`, `MessageNonce` from upstream to keep consistent.

Reply:
`LaneId` and `MessageNonce` have no definition of its abi decode/encode from upstream. is it better to keep the params as the primitive type?

Reply:
What if these two primitive updates in upstream, we also need to update the params here. But, When somebody global replaces the type, it's very easy to ignore the param types here.  No need to create a new type of defination.

> The MessageNonce if introduced here, we also deal with it as u64.

Yeah, In essence, they are the same thing.